kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/impl/AbstractInternalKrbClient.java

```diff
@@ -170,11 +171,22 @@ public SgtTicket requestSgt(KOptions requestOptions) throws KrbException {
                     tgsRequest = new TgsRequestWithTgt(context, sgtTicket);
                 }
             }
+        } else {
+            //This code is for the no-tgt case but works only with CLIENT_PRINCIPAL option
+            //Should be expanded later to encompass more use-cases
+            String clientPrincipalString = (String) requestOptions.getOptionValue(KrbOption.CLIENT_PRINCIPAL);
+            if (clientPrincipalString != null) {
+                clientPrincipalName = new PrincipalName(clientPrincipalString);
+            }
         }
-
+        
         tgsRequest.setServerPrincipal(serverPrincipalName);
         tgsRequest.setRequestOptions(requestOptions);
-        return doRequestSgt(tgsRequest);
+        SgtTicket sgtTicket = doRequestSgt(tgsRequest);
+        if(clientPrincipalName!=null){
```

Comment:
Please put spaces around brackets here.

Reply:
Gosh, I hate formatting :-) .. I'll fix this